engine/clients/pgvector/configure.py

```diff
@@ -0,0 +1,56 @@
+import psycopg2
+from psycopg2.extras import RealDictCursor
+
+from benchmark.dataset import Dataset
+from engine.base_client import IncompatibilityError
+from engine.base_client.configure import BaseConfigurator
+from engine.base_client.distances import Distance
+from engine.clients.pgvector.config import get_db_config
+
+
+class PgVectorConfigurator(BaseConfigurator):
+    DISTANCE_MAPPING = {
+        Distance.L2: "l2_norm",
+        Distance.COSINE: "cosine",
+        Distance.DOT: "dot_product",
+    }
+
+    def __init__(self, host, collection_params: dict, connection_params: dict):
+        super().__init__(host, collection_params, connection_params)
+        self.conn = psycopg2.connect(**get_db_config(host))
+        self.cursor = self.conn.cursor(cursor_factory=RealDictCursor)
+
+    def clean(self):
+        self.cursor.execute(
+            "DROP TABLE IF EXISTS items CASCADE;",
+        )
+        self.conn.commit()
+
+    def recreate(self, dataset: Dataset, collection_params):
+        if dataset.config.distance == Distance.DOT:
+            raise IncompatibilityError
+
+        self.cursor.execute(
+            f"""CREATE TABLE items (
+                id SERIAL PRIMARY KEY,
+                embedding vector({dataset.config.vector_size}) NOT NULL
+            );"""
+        )
+        self.cursor.execute(
+            "ALTER TABLE items ALTER COLUMN embedding SET STORAGE PLAIN"
+        )
```

Comment:
If we can recreate here, do we actually need the `init.sql`?

Reply:
Good point. I'll drop `init.sql`